commit f0988cf1484574e5c70e7ab263f6ee9e9545bb9b
Author: pppig236 <priv@pppig236.com>
Date:   Wed Dec 27 20:12:43 2023 -0500

    Revert "Check to support older ROCm versions (#963)"
    
    no thanks
    This reverts commit 198f14923b01fbf961a2a0be3ed83d251c17d73e.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5384287..68fc6df 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -119,26 +119,6 @@ execute_process(
   OUTPUT_VARIABLE hipcc_version_string)
 message(STATUS "hipcc version:    ${hipcc_version_string}")
 
-## Check for ROCm version
-execute_process(
-  COMMAND         bash "-c" "cat ${ROCM_PATH}/.info/version"
-  OUTPUT_VARIABLE rocm_version_string
-)
-string(REGEX MATCH "([0-9]+)\\.([0-9]+)\\.([0-9]+)" rocm_version_matches ${rocm_version_string})
-if (rocm_version_matches)
-    set(ROCM_MAJOR_VERSION ${CMAKE_MATCH_1})
-    set(ROCM_MINOR_VERSION ${CMAKE_MATCH_2})
-    set(ROCM_PATCH_VERSION ${CMAKE_MATCH_3})
-
-    message(STATUS "ROCm version: ${ROCM_MAJOR_VERSION}.${ROCM_MINOR_VERSION}.${ROCM_PATCH_VERSION}")
-
-    # Convert the version components to int for comparison
-    math(EXPR ROCM_VERSION "(10000 * ${ROCM_MAJOR_VERSION}) + (100 * ${ROCM_MINOR_VERSION}) + ${ROCM_PATCH_VERSION}")
-    add_definitions("-DROCM_VERSION=${ROCM_VERSION}")
-else()
-    message(WARNING "Failed to extract ROCm version.")
-endif()
-
 ### Check for hipEventDisableSystemFence support
 check_symbol_exists("hipEventDisableSystemFence" "hip/hip_runtime_api.h" HIP_EVENT_DISABLE_FENCE)
 
diff --git a/src/misc/argcheck.cc b/src/misc/argcheck.cc
index 7689acd..84f8a6f 100644
--- a/src/misc/argcheck.cc
+++ b/src/misc/argcheck.cc
@@ -14,11 +14,7 @@ static ncclResult_t CudaPtrCheck(const void* pointer, struct ncclComm* comm, con
     WARN("%s : %s %p is not a valid pointer", opname, ptrname, pointer);
     return ncclInvalidArgument;
   }
-#if ROCM_VERSION < 50500
-  if (attr.memoryType == cudaMemoryTypeDevice && attr.device != comm->cudaDev) {
-#else
   if (attr.type == cudaMemoryTypeDevice && attr.device != comm->cudaDev) {
-#endif
     WARN("%s : %s allocated on device %d mismatchs with NCCL device %d", opname, ptrname, attr.device, comm->cudaDev);
     return ncclInvalidArgument;
   }
